lib/semantics/semantics.h

```diff
@@ -135,6 +136,18 @@ class SemanticsContext {
   parser::Message &Say(parser::Message &&msg) {
     return messages_.Say(std::move(msg));
   }
+  template<typename... A>
+  void SayWithDecl(const Symbol &symbol, const parser::CharBlock &at,
+      parser::MessageFixedText &&msg, A &&... args) {
```

Comment:
This implicitly includes the symbol name as an argument to the message before `args`. That isn't obvious so it should be documented. If the caller wants something else as the first argument in the message, they have to do something special with the placeholders in the message. (I don't know what, but I know the feature is there.)

I think it would be better to only pass `args` to `Say` and require the caller to include `symbol.name()` explicitly in the position they want (if at all).

Reply:
Done.